Models/CertificateValidationRequest.swift

```diff
@@ -91,11 +95,13 @@ class CertificateValidationRequest {
     }
     
     private func computeLocalHash() {
-        localHash = sha256(data: certificate.file)
+        self.localHash = sha256(data: certificate.file)
         state = .fetchingRemoteHash
     }
     private func fetchRemoteHash() {
-        let transactionDataHandler = BlockchainInfoHandler(transactionId: transactionId)
+        
+        // todo: what is proper init pattern here?
+        let transactionDataHandler : TransactionDataHandler = TransactionDataHandler.create(chain: self.chain, transactionId: transactionId)
```

Comment:
Other style comments. You can omit the explicit type when it's obvious the right hand side only has one return type.